orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/PipelineExecutionImpl.java

```diff
@@ -150,12 +162,24 @@ public void setKeepWaitingPipelines(boolean keepWaitingPipelines) {
   }
 
   @JsonIgnore
+  @SuppressWarnings({"unchecked", "rawtypes"})
   public @Nonnull Map<String, Object> getContext() {
     return StageExecutionImpl.topologicalSort(stages)
         .map(StageExecution::getOutputs)
         .map(Map::entrySet)
         .flatMap(Collection::stream)
-        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue, (o, o2) -> o2));
+        .collect(
+            toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue,
+                (o, o2) -> {
+                  if (o instanceof Collection && o2 instanceof Collection) {
+                    return Stream.concat(((Collection) o).stream(), ((Collection) o2).stream())
+                        .distinct()
+                        .collect(Collectors.toList());
```

Comment:
Are we worried about this changing the type of collections in the context?  In particular, if we had a `Set` in the context this would now change it to a `List`.  This wouldn't matter (except maybe for performance) if everywhere we fetch it from the context we get it as a `Collection` but it might if there were places that we got it out expecting a `Set`.  Obviously I don't know that we want to over-complicate the solution and special case all collections, but just worth considering whether this could cause subtle (or maybe not that subtle if it's a cast error) bugs.

Reply:
I did consider the same thing, and actually started to implement it, but then threw it away because i didn’t want to complicate it too much, and I suspect this only ends up as Jain anyway, erasing any types anyway (but I’m not 100% sure).
I’m happy to reimplement it if you think it is a good idea.

Reply:
I think this should be fine the way you've written it, assuming nobody else can think of an edge case this would break.